### PR TITLE
moving PrettyDuration and PrettyDurationSpec to akka.util

### PR DESCRIPTION
**Pull request by lolski**
https://github.com/akka/akka/issues/18913

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
PLS BUILD

File: akka-actor/src/main/scala/akka/util/PrettyDuration.scala [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
  */
-package akka.testkit.metrics.reporter
+package akka.util
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: should we add a Java API also, since it's now public?

Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: :+1: a simple `def create(...)` will do (and document as `/** Java API */`)

Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: or?

```
object PrettyDuration {
   /** JAVA API */
   format(duration: Duration, includeNanos: Boolean, precision: Int): String =
```

Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: Good one, I concur.

Comment (lolski) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: Got it

Comment (lolski) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: Should we add a default argument, e.g. `def format(duration: Duration, includeNanos: Boolean = false, precision: Int = 4)`? In this example, both arguments are identical to the implementations of `PrettyPrintableDuration.pretty`

Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/PrettyDuration.scala]: It's Java API, can't have default arguments :-)
But yes, let's do an overload with `def format(duration) = format(duration, includeNanos = false, precision = 4)`

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
LGTM, please squash into 1 commit with message according to our guidelines: https://github.com/akka/akka/blob/master/CONTRIBUTING.md#creating-commits-and-writing-commit-messages
So in this case `+act #18913 moved PrettyDuration to akka-actor`

Thanks!

**Comment by lolski** [comment]
Done!

**Comment by ktoso** [comment]
LGTM, thanks; will merge after validation.

**Comment by drewhk** [comment]
Failure seems to be unrelated

**Comment by johanandren** [comment]
PLS BUILD

**Comment by lolski** [comment]
Added a few lines of docs for the function as @patriknw suggested

**Comment by ktoso** [comment]
Looks good to me, thanks!

**Pull request merged**